CTS1.rst

```diff
@@ -0,0 +1,666 @@
+
+.. note::
+
+  The specification does not prevent adding more fine-grained authentication, such as a token level authentication.
+
+Operator can transfer any amount of any token type of the owner
+---------------------------------------------------------------
+
+An operator of an address can transfer any amount of any token type owned by the address.
+An alternative approach could be to scope the operators per token type and the owner could then add the operator for every token type to achieve the same.
+Although it is a more flexible approach in terms of functionality, the complexity will require more of the contract implementation and the general interaction by off-chain integrations and other smart contracts, which in turn would increase the energy cost.
+
+However, if a more fine grained authentication system is needed it can still exist next to the operators.
+
+Receive hook function
+---------------------
+
+The specification requires a token receive hook to be called on a smart contract receiving tokens, this will in some cases prevent mistakes such as sending tokens to smart contracts which do not define behavior for receiving tokens.
+These token could then be lost forever.
+
+The reason for this not being optional is to allow other smart contracts which integrate with a token smart contract to rely on this for functionality.
+
+.. warning::
+
+  The smart contract receive hook function could be called by any smart contract and it is up to the integrating contract whether to trust the token contract.
+
+Receive hook function callback argument
+---------------------------------------
+
+The name of the receive hook function called on a smart contract receiving tokens is supplied as part of the parameter.
+This allows for a smart contract integrating with a token smart contract to have multiple hooks and leave it to the caller to know which hook they want to trigger.
+
+Another technical reason is that the name of the smart contract is part of the smart contract receive function name, which means the specification would include a requirement of the smart contract name for other to integrate reliably.
+
+No sender hook function
+-----------------------
+
+The FA2 token standard found on Tezos, allows for a hook function to be called on a smart contract sending tokens, such that the contract could reject the transfer on some criteria.
+This seems to only make sense, if some operator is transferring tokens from a contract, in which case the sender smart contract might as well contain the logic to transfer the tokens and trigger this directly.
+
+Explicit events for mint and burn
+---------------------------------
+
+In ERC20, ERC721 and ERC1155 they use a transfer event from or to the zero address to indicate mint and burn respectively, but since there are no such thing as the zero address on the Concordium blockchain these events are separate.
```

Comment:
```suggestion
ERC20, ERC721 and ERC1155 use a transfer event from or to the zero address to indicate mint and burn respectively, but since there are no such thing as the zero address on the Concordium blockchain these events are separate.
```